s3/src/main/java/io/ddf/s3/S3DDF.java

```diff
@@ -0,0 +1,135 @@
+package io.ddf.s3;
+
+import io.ddf.DDF;
+import io.ddf.DDFManager;
+import io.ddf.content.Schema;
+import io.ddf.content.SqlResult;
+import io.ddf.datasource.DataFormat;
+import io.ddf.exception.DDFException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by jing on 12/2/15.
+ */
+public class S3DDF extends DDF {
+    // Whether this ddf has header.
```

Comment:
Minor style issue, maybe add a newline after each variable declaration for readability.